core-plugins/src/test/java/co/cask/hydrator/plugin/batch/joiner/JoinerTestRun.java

```diff
@@ -612,64 +497,81 @@ private void verifyOuterJoinWithoutRequiredInputs(Schema outputSchema, TimeParti
     Assert.assertEquals(8, actual.size());
     org.apache.avro.Schema avroOutputSchema = new org.apache.avro.Schema.Parser().parse(outputSchema.toString());
 
-    GenericRecord bobRecord1 = new GenericRecordBuilder(avroOutputSchema)
+    GenericRecord humtumRecord = new GenericRecordBuilder(avroOutputSchema)
+      .set("film_id", "4")
+      .set("film_name", "humtum")
+      .set("renamed_actor", null)
+      .set("renamed_category", null)
+      .build();
+
+    GenericRecord sultanRecord = new GenericRecordBuilder(avroOutputSchema)
+      .set("film_id", null)
+      .set("film_name", null)
+      .set("renamed_actor", null)
+      .set("renamed_category", "comedy")
+      .build();
+
+    Set<GenericRecord> expected = ImmutableSet.of(getBobRecord1(avroOutputSchema),
+                                                  getBobRecord2(avroOutputSchema),
+                                                  getAlexRecord1(avroOutputSchema),
+                                                  getAlexRecord2(avroOutputSchema),
+                                                  getCathieRecord1(avroOutputSchema),
+                                                  getAvatarRecord1(avroOutputSchema),
+                                                  humtumRecord, sultanRecord);
+    Assert.assertEquals(expected, actual);
+  }
+
+  GenericRecord getBobRecord1(org.apache.avro.Schema avroOutputSchema) {
```

Comment:
As we use different `avroOutputSchema` depending on type of the join (For outer joins, we use nullable fields), we need new instances of records.